src/utils/exportUtils/Exporter.ts

```diff
@@ -0,0 +1,7 @@
+import { MatrixEvent } from "matrix-js-sdk/src/models/event";
+import Room from 'matrix-js-sdk/src/models/room';
+
+export abstract class Exporter {
+    constructor(protected res: MatrixEvent[], protected room: Room) {}
+    abstract export(): Promise<void>
```

Comment:
```suggestion
    abstract export(): Promise<void>;
```

Reply:
this might want to return the actual file/files to be downloaded maybe

Reply:
Sure